typed-function.js

```diff
@@ -91,61 +99,128 @@
 
     var anyType = {
       name: 'any',
-      test: ok
+      test: ok,
+      isAny: true
     }
 
+    // Data structures to track the types. As these are local variables in
+    // create(), each typed universe will get its own copy, but the variables
+    // will only be accessible through the (closures of the) functions supplied
+    // as properties of the typed object, not directly.
+    // These will be initialized in clear() below
+    var typeMap; // primary store of all types
+    var typeList; // Array of just type names, for the sake of ordering
+
     // types which need to be ignored
     var _ignore = [];
 
-    // type conversions
-    var _conversions = [];
+    // And similar data structures for the type conversions:
+    var nConversions = 0;
+    // the actual conversions are stored on a property of the destination types
 
-    // This is a temporary object, will be replaced with a typed function at the end
+    // This is a temporary object, will be replaced with a function at the end
     var typed = {
-      types: _types,
-      conversions: _conversions,
       ignore: _ignore,
       createCount: 0
     };
 
     /**
-     * Find the test function for a type
-     * @param {String} typeName
-     * @return {TypeDef} Returns the type definition when found,
-     *                    Throws a TypeError otherwise
+     * Takes a type name or a type object and returns the corresponding
+     * officially registered type object for that type.
+     * @param {string | Type | TypeDef} typeSpec
+     * @returns {TypeDef} type
      */
-    function findTypeByName (typeName) {
-      var entry = findInArray(typed.types, function (entry) {
-        return entry.name === typeName;
-      });
-
-      if (entry) {
-        return entry;
+    function findType (typeSpec) {
+      let type = typeof typeSpec === 'string' ?
+        typeMap.get(typeSpec) :
+        typeMap.get(typeSpec.name)
+      if (type) return type;
+      // Remainder is error handling
+      var name = typeof typeSpec === 'string' ? typeSpec : typeSpec.name;
```

Comment:
> Disallowing a typespec argument as you suggest below will also do that, so I will pursue that and fix the calls as needed.

👍 

> I have been avoiding let and const because frankly I now have no idea what's allowed in IE11. But if you tell me let and const are generally ok I will gladly change many vars to them

Yes I'm not totally sure about that too. According to https://caniuse.com/?search=let, IE11 supports `const` and `let` except in for loops. So besides that we can just use them. We'll need to be a bit careful and test if all still works on IE in the end.

Reply:
OK, used let and const as much as possible from my understanding of your statement. Fingers crossed it works on IE11, at least until mid-June.

Reply:
😂 yes